azDevOps/azure/templates/v2/scripts/inject-secrets-from-env-var.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+secretfolderpath=$1
+
+cd $secretfolderpath
+echo "Scanning for files in $secretfolderpath"
+echo "Found these files:"
+ls -lp1
+
+for file in $(find -type f); do
+
+    envVar=$(echo "${file#"./"}" | tr / _);
+    eval "tmpval=\"\$$envVar\""
```

Comment:
We need to add extra logic to this script to handle case sensitive values
Currently, need to be an exact match between the filename and the ENV VAR name, that is generally uppercase
We should look for an ENV var as UPPERCASE instead of the name of the file

Reply:
Also, by default, AzDO set library as UPPERCASE when defining an ENV VAR

Reply:
tbh, this was a blanket copy of your previous as is working currently. but there is a bit of work to do on this I think. 
Also, there is nothing wrong with variables being caps as they are basically globals/environment variables and the convention is that they are always upper case

Reply:
This whole secret management solution is a pita... It's got so many layers and doesn't support multiple variables of the same type.

Ideally we'd come up with a better solution.

For example, to add a variable to a dotnet app you need to make a script that extracts the variable, add code to both the build and deploy to extract that variable.

Then you have to make sure it gets passed to this script and the name has to match the files on the file system for Kustomise (hence why uppercase isn't a good idea here).

Then after that to add it to an app you have to create the secret files in each env in the repo to deploy to, _then_ alter each Kustomise file to override it.

THEN you have to go to the deployment yaml and ensure the variable is mapped to an env var in the app (completely separate from the variable defined in the base repo) based on the k8s secret that Kustomise generates. 

All to add a single variable to a build!